redisgears_core/src/stream_reader.rs

```diff
@@ -17,6 +17,11 @@ use std::time::{SystemTime, UNIX_EPOCH};
 
 use crate::RefCellWrapper;
 
+pub type RecordAcknowledgeCallback = dyn Fn(&[u8], u64, u64);
+pub type StreamReaderCallback<T> =
+    dyn Fn(&[u8], Option<RedisModuleStreamID>, bool) -> Result<Option<T>, String> + Sync + Send;
+pub type StreamTrimmerCallback = dyn Fn(&[u8], RedisModuleStreamID) + Sync + Send;
```

Comment:
I want to re-write this entire section anyway, but I guess we can start with those small improvements.